src/libraries/System.Private.DataContractSerialization/src/System/Xml/XmlBufferReader.cs

```diff
@@ -964,10 +964,10 @@ public ulong GetUInt64(int offset)
             => (ulong)GetInt64(offset);
 
         public float GetSingle(int offset)
-            => ReadRawBytes<float>(offset);
+            => BinaryPrimitives.ReadSingleLittleEndian(_buffer.AsSpan(offset, 4));
 
         public double GetDouble(int offset)
```

Comment:
There are also GetSingle/GetDouble methods without the "int" parameter

Reply:
Thanks, I had missed those.   Now added.